core/space/services/services_fs.go

```diff
@@ -530,7 +542,12 @@ func (s *Space) addFile(ctx context.Context, sourcePath string, targetPath strin
 
 	_, fileName := filepath.Split(sourcePath)
 
-	targetPathBucket := targetPath + "/" + fileName
+	var targetPathBucket string
+	if targetPath == "" {
```

Comment:
what of situation where a user specifies targetPath as `"/"` and not empty string. The `b.UploadFile()` already handles this edge case but how does it affect this if statement here? Maybe you should also check for paths only having a path separator.

Reply:
Fixed in [8ccc69f](https://github.com/FleekHQ/space-daemon/pull/180/commits/8ccc69f2f66636b1462fbe89caf17f0d6b2252ff).